### PR TITLE
Feature/publish workflow images to GitHub package registry

### PR DESCRIPTION
**Pull request by jshilts657**
Not meant to be an official PR just set this up to easily review changes we both made.

Can close and re-open a new one under your name when it's ready.

### CODE REVIEW

File: .github/workflows/current-workflow.yml [hunk 1]
```diff
@@ -120,3 +187,61 @@ jobs:
...
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract tag data for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
```
Comment (jshilts657) [inline .github/workflows/current-workflow.yml]: @rzheng2019 
Update to match

File: .github/workflows/current-workflow.yml [hunk 2]
```diff
@@ -120,3 +187,61 @@ jobs:
...
+
+      - name: Build and push the Docker Image
+        id: push
+        uses: docker/build-push-action@v5
+        with:
+          context: ./code/frontend
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}_frontend_${{ github.sha }}
+          labels: ${{ steps.meta.outputs.labels }}
```
Comment (jshilts657) [inline .github/workflows/current-workflow.yml]: might need to remove? or investigate

File: .github/workflows/current-workflow.yml [hunk 3]
```diff
@@ -120,3 +187,61 @@ jobs:
...
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push the Docker Image
+        id: push
+        uses: docker/build-push-action@v5
+        with:
+          context: ./code/frontend
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}_frontend_${{ github.sha }}
```
Comment (jshilts657) [inline .github/workflows/current-workflow.yml]: update to match the way it was done in backend

File: .github/workflows/current-workflow.yml [hunk 4]
```diff
@@ -120,3 +187,61 @@ jobs:
...
+          labels: ${{ steps.meta.outputs.labels }}
+        
+      - name: Generate artifact attestation
+        uses: actions/attest-build-provenance@v1
+        with:
+          subject-name: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME}}
```
Comment (jshilts657) [inline .github/workflows/current-workflow.yml]: here too

File: .github/workflows/current-workflow.yml
Comment (jshilts657) [inline .github/workflows/current-workflow.yml:1]: Update   Docker-and-Node-Service: section as well

File: .github/workflows/current-workflow.yml (hunk 1 above)
Comment (rzheng2019) [reply .github/workflows/current-workflow.yml]: Frontend has been updated with the same convention as backend.

File: .github/workflows/current-workflow.yml (hunk 3 above)
Comment (rzheng2019) [reply .github/workflows/current-workflow.yml]: Frontend has been updated with the same convention as backend.

File: .github/workflows/current-workflow.yml (hunk 4 above)
Comment (rzheng2019) [reply .github/workflows/current-workflow.yml]: Frontend has been updated with the same convention as backend.

File: .github/workflows/current-workflow.yml
Comment (rzheng2019) [reply .github/workflows/current-workflow.yml:1]: Frontend has been updated with the same convention as backend.

**Comment by jshilts657** [comment]
This PR is ready to be merged in my opinion
Create a PR under your name when you are ready.
Make sure you merge dev into your branch before you merge

File: .github/workflows/current-workflow.yml [hunk 5]
```diff
@@ -31,25 +39,86 @@ jobs:
...
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.REPO_NAME }}/${{ env.BACKEND_IMAGE_NAME }}
+
+      - name: Build and push the Docker Image
+        id: push
+        uses: docker/build-push-action@v5
+        with:
+          context: ./code/backend
+          push: true
+          tags: ${{ env.REGISTRY }}/${{ env.REPO_NAME }}/${{ env.BACKEND_IMAGE_NAME }}:latest
```
Comment (jshilts657) [inline .github/workflows/current-workflow.yml]: instead of latest lets try :${{ github.ref_name }}

**Pull request closed without merging**